Configure

```diff
@@ -1575,6 +1584,8 @@ unless ($disabled{afalgeng}) {
 
 push @{$config{openssl_feature_defines}}, "OPENSSL_NO_AFALGENG" if ($disabled{afalgeng});
 
+push @{$config{openssl_feature_defines}}, "OPENSSL_DEVCRYPTO_DYNAMIC" if (!$disabled{"devcryptoeng-dynamic"});
```

Comment:
Hmm...  I think it will be confusing if we have a mixture of "yes" and "no" macros for this sort of thing...  May I suggest doing like the others and define `OPENSSL_NO_DEVCRYPTOENG_DYNAMIC` if it's disabled?

Reply:
(either way, looking at already established patterns, it should be `DEVCRYPTOENG` rather than `DEVCRYPTO`)

Reply:
My logic was that defaults should not have anything defined (afalg is a default, isn't it?).  I'll change it anyway.

Reply:
I get it, but our pattern is that we disable thing (by default or not doesn't matter), and that "negative" view on options is reflected in the corresponding macros...  or we un-disabled things (when they are enabled, by default or not), which is reflected with the absence of the negative macros.

(between you and me, I would have much preferred "positive" macros, but that would be a huuuuge change that I don't think we're ready for)

Reply:
> (between you and me, I would have much preferred "positive" macros, but that would be a huuuuge change that I don't think we're ready for)

Strong agreement that it would have been better, and also that sadly it's too late. Even back when I pruned down all the options for 1.1.0 it was too late.

Mixing no-xxx with potential yes-yyy is bad, even if it means that sometimes the ifdef maze gets a little tortured.

Reply:
It turns out this line is not needed when using `OPENSSL_NO_DEVCRYPTOENG_DYNAMIC`.  It is defined in `Configure: 1210`
```
my $macro = $disabled_info{$what}->{macro} = "OPENSSL_NO_$WHAT";
```
once I took it off the exception list in `1200..1202`.